gap/base/recognition.gi

```diff
@@ -174,13 +174,35 @@ InstallGlobalFunction( EmptyRecognitionInfoRecord,
     ri!.pr := ProductReplacer(GeneratorsOfGroup(H));
     ri!.gensHmem := GeneratorsWithMemory(GeneratorsOfGroup(H));
     ri!.prodrep := ProductReplacer(ri!.gensHmem, rec( maxdepth := 400 ));
+    # The purpose of the following components is to use the components randr
+    # and rando as the central places where random elements and their orders
+    # can be stored.
+    # RandomElm and RandomElmOrd both take a string stamp as second input. The
+    # string stamp can be used to request a random elment or order for a
+    # specific computation. If say a computation which used stamp := "A" has
+    # already computed random elements or orders, then RandomElm and
+    # RandomElmOrd will reuse these if called with stamp := "B".
+    #
+    # Notice that RandomElm is by default stored in the component
+    # H!.pseudorandomfunc. A method for PseudoRandom(H) is installed such that
+    # it calls RandomElm(ri, "PseudoRandom", false).
+    #
+    # randr stores the random elements computed by RandomElm and RandomElmOrd
     ri!.randr := EmptyPlist(100);
+    # If the order of a random element is computed by RandomElmOrd or if the
+    # order of an entry of randr is computed by GetElmOrd, then the order is
+    # saved into rando.
     ri!.rando := EmptyPlist(100);
-    ri!.randp := EmptyPlist(100);
+    # randrpt stores for each stamp how often it was used to generate a random
+    # element.
     ri!.randrpt := rec();
+    # randopt stores for each stamp how often it was used to generate a random
+    # order.
     ri!.randopt := rec();
-    ri!.randppt := rec();
```

Comment:
@aniemeyer wrote this to us:
> I had a look at this again.
> 
> As far as I can tell  these functions are not used anywhere. I do not know who wrote them and
> why. They are not part of the classical recognition code.
> 
> However, my guess is the following: 
> There are many algorithms that use various types of information about random elements, such
> as their order, what type of ppd elements they are etc. At the moment the various algorithms
> called within the recog tree all do their own thing. However, it might be useful if certain information
> would be stored globally. So if one method already computed the orders of elements, another method
> might as well use the computed orders without producing new random elements to compute their orders.
> 
> It seems to me that the functions you found were intended to be run on every random element generated
> and the outcomes to be used by any method applied to the group in question.
> 
> However, this is just a guess, so it may be incorrect.

Reply:
Ah, my bad sorry. I'll try and be more careful with how the diffs will appear to you. I actually didn't delete it, but I commented it out _and moved it_ two lines down. I'm in favor of keeping the code for the ppd elements. The function `GetElmPpd` stores things like characterstic polynomials and their factors. I can imagine that not having to recompute that may save us some time. I'm opening a new issue for that.